src/plugin/luapi_application.h

```diff
@@ -305,13 +385,381 @@ static int applib_changeToolColor(lua_State* L) {
+    Control* control = plugin->getControl();
+    size_t pageId = luaL_checkinteger(L, 1);
+    const size_t first = 1;
+    const size_t last = control->getDocument()->getPageCount();
+    control->firePageSelected(std::clamp(pageId, first, last) - 1);
+
+    return 1;
+}
+
+/**
+ * Sets the width and height of the current page in pt = 1/72 inch either relatively or absolutely (by default)
+ *
+ * Example 1: app.setPageSize(595.275591, 841.889764)
+ * makes the current page have standard (A4 paper) width and height (absolute mode)
+ *
+ * Example 2: app.setPageSize(0, 14.17*6, true)
+ * adds 14.17*6 pt = 3cm to the height of the page (relative mode)
+ **/
+static int applib_setPageSize(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    Document* doc = control->getDocument();
+    PageRef const& page = control->getCurrentPage();
+
+    if (!page) {
+        luaL_error(L, "No page!");
+    }
+
+    double width = luaL_checknumber(L, 1);
+    double height = luaL_checknumber(L, 2);
+
+    bool relative = false;
+    if (lua_isboolean(L, 3)) {
+        relative = lua_toboolean(L, 3);
+    }
+
+    if (relative) {
+        width += page->getWidth();
+        height += page->getHeight();
+    }
+
+    if (width > 0 and height > 0) {
+        doc->lock();
+        Document::setPageSize(page, width, height);
+        doc->unlock();
+    }
+
+    size_t pageNo = doc->indexOf(page);
+    if (pageNo != npos && pageNo < doc->getPageCount()) {
+        control->firePageSizeChanged(pageNo);
+    }
+
+    return 1;
+}
+
+/**
+ * Sets the current layer of the current page as indicated and updates visibility if specified (by default it does not)
+ * Displays an error message, if the selected layer does not exist
+ *
+ * Example 1: app.setCurrentLayer(2, true)
+ * makes the second (non-background) layer the current layer and makes layers 1, 2 and the background layer visible, the
+ *others hidden
+ *
+ * Example 2: app.setCurrentLayer(2, false)
+ * makes the second (non-background) layer the current layer and does not change visibility
+ **/
+static int applib_setCurrentLayer(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    PageRef const& page = control->getCurrentPage();
+
+    if (!page) {
+        luaL_error(L, "No page!");
+    }
+
+    size_t layerCount = page->getLayerCount();
+    size_t layerId = luaL_checkinteger(L, 1);
+
+    if (layerId < 0 || layerId > layerCount) {
+        luaL_error(L, "No layer with layer ID %d", layerId);
+    }
+
+    bool update = false;
+    if (lua_isboolean(L, 2)) {
+        update = lua_toboolean(L, 2);
+    }
+
+    control->getLayerController()->switchToLay(layerId, update);
+
+    return 1;
+}
+
+/*
+ * Sets the visibility of the current layer
+ *
+ * Example: app.setLayerVisibility(true)
+ * makes the current layer visible
+ */
+static int applib_setLayerVisibility(lua_State* L) {
```

Comment:
I think this would be more useful if you could supply an optional layer number parameter.

Reply:
On the other hand, one can simply use `app.setCurrentLayer` followed by `app.setLayerVisibility` to achieve the same thing (one may need to set the current layer back at the end though). In general I tried to keep the API simple and let the Lua code set (or iterate through) layers and pages. See also e.g. `applib_setPageSize`, where also no page number is specified.
If you insist, I can change it, of course.

Reply:
Fair enough. We could easily add an extra parameter if someone complains that the API is clunky.